src/sonic-yang-models/yang-models/sonic-extension.yang

```diff
@@ -7,7 +7,54 @@ module sonic-extension {
 
     description "Extension yang Module for SONiC OS";
 
+    revision 2021-02-23 {
+        description "Adding extensions as per Sonic Yang Guidelines";
+    }
+
     revision 2019-07-01 {
         description "First Revision";
     }
+
+    /* For complete guide of using these extensions in SONiC yangs, refer
+    SONiC yang guidelines at
+    https://github.com/Azure/SONiC/blob/master/doc/mgmt/SONiC_YANG_Model_Guidelines.md */
+
+    extension db-name {
+        description "DB name, e.g. APPL_DB, CONFIG_DB";
+        argument "value";
+    }
+
+    extension key-delim {
+        description "Key delimeter, e.g. - |, :";
+        argument "value";
+    }
+
+    extension key-pattern {
+        description "Key pattern, e.g. - ACL_RULE|{aclname}|{rulename}";
```

Comment:
Why this extension is needed.

Reply:
As per this below example, this extension is required to define patterns of key.
       sonic-ext:key-pattern "QUEUE|({ifname},)*|{qindex}";

```
    "QUEUE": {
        "Ethernet0,Ethernet4,Ethernet8,Ethernet12,Ethernet16,Ethernet120,Ethernet124|0": {
            "scheduler": "[SCHEDULER|scheduler.1]"
        },
        "Ethernet0,Ethernet4,Ethernet8,Ethernet12,Ethernet16,Ethernet100,Ethernet104,Ethernet124|3-4": {
            "wred_profile": "[WRED_PROFILE|AZURE_LOSSLESS]",
            "scheduler": "[SCHEDULER|scheduler.0]"
        }
    }
```